### PR TITLE
Max/float53

### PR DESCRIPTION
**Pull request by max-hoffman**
adds support and tests for `FLOAT(prec)`, which will help with pandas table creation.

### CODE REVIEW

**Review by zachmu — APPROVED** [review]
LGTM but @Hydrocharged should confirm this is the correct semantics for the float type. He understands the types better than I do.

File: go/vt/sqlparser/sql.y [hunk 1]
```diff
@@ -1320,6 +1320,12 @@ float_length_opt:
   {
     $$ = LengthScaleOption{}
   }
+| '(' INTEGRAL ')'
```
Comment (Hydrocharged) [inline go/vt/sqlparser/sql.y:1323]: `float_length_opt` is a duplicate of `decimal_length_opt` with this, making the addition redundant, so we can use `decimal_length_opt` instead. It also allows `DOUBLE(n)`, which is not supported.

**Review by Hydrocharged — CHANGES_REQUESTED** [review]
`FLOAT(n, m)` is deprecated behavior, from pre-`DECIMAL` days, but we can still parse it. We should error on it in the engine though.

File: go/vt/sqlparser/sql.y [hunk 2]
```diff
@@ -3345,6 +3351,12 @@ convert_type:
     $$.Length = $2.Length
     $$.Scale = $2.Scale
   }
+| FLOAT_TYPE float_length_opt
```
[the remark this answers is not in the document]
Comment (max-hoffman) [reply go/vt/sqlparser/sql.y]: cool, i think I got it right this time. had to re-read the tutorial a few times, syntax was a bit perplexing yesterday

**Comment by max-hoffman** [comment]
> We should error on it in the engine though.

@Hydrocharged What do you mean by this? I saw the deprecation in the mysql docs, but I need it to not error in practice

**Comment by Hydrocharged** [comment]
> 
> 
> > We should error on it in the engine though.
> 
> @Hydrocharged What do you mean by this? I saw the deprecation in the mysql docs, but I need it to not error in practice

Do we need `FLOAT(n)` functionality only? Or we need `FLOAT(n, m)` as well? I was only speaking to `FLOAT(n, m)` regarding the error.

**Comment by max-hoffman** [comment]
Ok got it, do you have any suggestions for implementing that other than of defining a separate `length_op`?

**Review by Hydrocharged — APPROVED** [review]
Covered it all in a quick chat so we're good to go!

**Pull request merged**